Applications/Spire/Source/UiViewer/MarketComboBoxTestWidget.cpp

```diff
@@ -0,0 +1,26 @@
+#include "Spire/UiViewer/MarketComboBoxTestWidget.hpp"
+#include <QHBoxLayout>
+#include "Nexus/Definitions/DefaultMarketDatabase.hpp"
+#include "Spire/Ui/MarketComboBox.hpp"
+#include "Spire/Ui/CustomQtVariants.hpp"
+
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+MarketComboBoxTestWidget::MarketComboBoxTestWidget(QWidget* parent)
+    : ComboBoxAdapter(parent) {
+  auto layout = new QHBoxLayout(this);
+  layout->setContentsMargins({});
+  auto database = GetDefaultMarketDatabase();
+  auto combo_box = new MarketComboBox(database, this);
+  layout->addWidget(combo_box);
+  combo_box->connect_selected_signal([=] (auto value) {
+    m_selected_signal(QVariant::fromValue<MarketToken>(value));
+  });
+}
+
+connection MarketComboBoxTestWidget::connect_selected_signal(
```

Comment:
What will ever call this method?

Reply:
The ComboBoxTestWidget calls it to connect the selection signal.

Reply:
It should be marked override, though.